src/animation/offline/tools/import2ozz_anim.cc

```diff
@@ -385,19 +388,19 @@ bool ImportAnimations(const Json::Value& _config, OzzImporter* _importer,
     }
 
     bool matched = false;
-    for (size_t j = 0; success && j < import_animation_names.size(); ++j) {
+    for (size_t j = 0; j < import_animation_names.size(); ++j) {
       const char* animation_name = import_animation_names[j].c_str();
       if (!strmatch(animation_name, clip_match)) {
         continue;
       }
 
       matched = true;
-      success = ProcessAnimation(*_importer, animation_name, *skeleton,
+      bool anisuccess = ProcessAnimation(*_importer, animation_name, *skeleton,
```

Comment:
success variable is not updated in this loop, based on anisuccess. That means that the application will exit with a success which isn't expected (as discussed).

Reply:
It my mistake, it should return failed when one of the animation failed. And I will add some unit test you mention above with this fix.